src/record_mixing.c

```diff
@@ -1018,57 +1018,59 @@ static void Task_DoRecordMixing(u8 taskId)
 
 // New Emerald functions
 
-static void sub_80E8110(struct Apprentice *dst, struct Apprentice *src)
+static void GetSavedApprentices(struct Apprentice *dst, struct Apprentice *src)
 {
     s32 i, id;
-    s32 var_2C, var_28, var_24, r8;
+    s32 apprenticeSaveId, oldPlayerApprenticeSaveId;
+    s32 numOldPlayerApprentices, numMixApprentices;
 
     dst[0].playerName[0] = EOS;
     dst[1].playerName[0] = EOS;
 
     dst[0] = src[0];
 
-    var_28 = 0;
-    var_24 = 0;
-    var_2C = 0;
-    r8 = 0;
+    oldPlayerApprenticeSaveId = 0;
+    numOldPlayerApprentices = 0;
+    apprenticeSaveId = 0;
+    numMixApprentices = 0;
     for (i = 0; i < 2; i++)
     {
-        id = ((i + gSaveBlock2Ptr->playerApprentice.field_B2_1) % 3) + 1;
+        id = ((i + gSaveBlock2Ptr->playerApprentice.saveId) % 3) + 1;
```

Comment:
I think this `% 3` is supposed to be `APPRENTICE_COUNT - 1`

Reply:
Yeah, `saveId` is used as an index into `gSaveBlock2Ptr->apprentices`. Updated